### PR TITLE
CORTX-30044 Redirect hare log messages to stdout/stderr

### PR DESCRIPTION
**Pull request by seagate-sarang-sawant**
Problem: Redirect Error and Warning messages of Hare processes to stdout/stderr

Solution: Add stream handler to processes missing it. Stream handlers would
have INFO level logging with process id or thread added in log message
depending on context. Log level of some messages are changed to info
from debug.

Signed-off-by: Sarang Sawant <sarang.sawant@seagate.com>

### CODE REVIEW

**Comment by pavankrishnat** [comment]
retest this please

**Review by pavankrishnat — APPROVED** [review]

**Comment by mssawant** [comment]
@seagate-sarang-sawant, please fix the commit message with respect to following points,
```
CORTX-30044 Redirect hare log messages to stdout/stderr

Problem: Redirect Error and Warning messages of Hare processes to stdout/stderr

Solution: Add stream handler to processes missing it. Stream handlers would
have INFO level logging with process id or thread added in log message
depending on context. Log level of some messages are changed to info
from debug.
```
1. Avoid capitalization post `:`
2. Itemize the problem and solution descriptions in case of multiple reasons.
3. Need not duplicate the summary in problem statement if there's nothing additional to describe with respect to the problem.

File: cfgen/cfgen [hunk 1]
```diff
@@ -41,10 +43,10 @@ class DiskRef():
 
 __version__ = '0.14'
 
-faultToleranceInfo = NamedTuple('faultToleranceInfo', [
-                            ('max_drives_per_node', int),
-                            ('max_drives_per_ctrl', int),
-                            ('min_drives_per_node', int)])
+faultToleranceInfo = NamedTuple('faultToleranceInfo',
```
Comment (mssawant) [inline cfgen/cfgen:46]: Why was this change needed? kindly avoid the unrelated changes.

File: cfgen/cfgen [hunk 2]
```diff
@@ -264,10 +266,13 @@ def setup_logging(opts: Any) -> None:
...
         verbosity = logging.DEBUG
-
-    logging.basicConfig(level=verbosity,
-                        handlers=handlers,
-                        format='%(asctime)s [%(levelname)s] %(message)s')
+    log_format = "%(asctime)s %(name)s %(process)d [%(levelname)s] " \
+                 "%(message)s"
+    formatter = logging.Formatter(fmt=log_format)
+    LOG.setLevel(verbosity)
+    for hndlr in handlers:
```
Comment (mssawant) [inline cfgen/cfgen:273]: I think `handler` or `h` is more readable and less prone to mistakes.

File: cfgen/cfgen [hunk 3]
```diff
@@ -398,7 +403,7 @@ def enrich_cluster_desc(desc: Dict[str, Any], mock_p: bool) -> None:
         if all(key in node.keys()
                for key in ('processorcount', 'memorysize_mb')):
 
-            logging.debug(
+            LOG.debug(
```
Comment (mssawant) [inline cfgen/cfgen:406]: I think this is `info`, I don't see `debug` specific information here.

File: cfgen/cfgen [hunk 4]
```diff
@@ -469,11 +474,11 @@ def validate_cluster_desc(desc: Dict[str, Any]) -> None:
         validate_profiles_desc(desc['profiles'], desc['pools'])
     if 'fdmi_filters' in desc and desc['fdmi_filters'] is not None:
         validate_fdmi_filters_desc(desc['fdmi_filters'], desc['nodes'])
-    logging.debug('Validations passed')
+    LOG.debug('Validations passed')
```
Comment (mssawant) [inline cfgen/cfgen:477]: I think this is good info to understand the various validation stages performed.

File: cfgen/cfgen [hunk 5]
```diff
@@ -469,11 +474,11 @@ def validate_cluster_desc(desc: Dict[str, Any]) -> None:
...
 
 
 def validate_nodes_desc(nodes_desc: List[Dict[str, Any]]) -> None:
-    logging.debug('Validating node descriptions')
+    LOG.debug('Validating node descriptions')
```
Comment (mssawant) [inline cfgen/cfgen:481]: Same as above, seems info.

File: cfgen/cfgen [hunk 6]
```diff
@@ -561,7 +566,7 @@ Make sure the value of data_iface in the CDF is correct.""")
 
 
 def validate_pools_desc(pools_desc: List[Dict[str, Any]]) -> None:
-    logging.debug('Validating pool descriptions')
+    LOG.debug('Validating pool descriptions')
```
Comment (mssawant) [inline cfgen/cfgen:569]: info?

File: cfgen/cfgen [hunk 7]
```diff
@@ -739,7 +744,7 @@ def validate_facter(hostname: str) -> None:
 
 def get_facts(hostname: str, mock_p: bool, *args: str) -> Dict[str, Any]:
     if mock_p:
-        logging.debug('Node facts will be fabricated because of --mock flag')
+        LOG.debug('Node facts will be fabricated because of --mock flag')
```
Comment (mssawant) [inline cfgen/cfgen:747]: This again seems like information.

File: hax/hax/util.py [hunk 8]
```diff
@@ -62,13 +62,14 @@
 MotrConsulProcInfo = NamedTuple('MotrConsulProcInfo', [('proc_status', str),
                                                        ('proc_type', str)])
 
-MotrConsulProcStatus = NamedTuple('MotrConsulProcStatus', [(
-                                        'consul_svc_status', str),
-                                        ('consul_motr_proc_status', str)])
+MotrConsulProcStatus = NamedTuple('MotrConsulProcStatus',
```
Comment (mssawant) [inline hax/hax/util.py:65]: Through out Hare code we follow https://peps.python.org/pep-0008/#indentation guide lines. So opening brackets must be on the same line.

File: hax/hax/util.py [hunk 9]
```diff
@@ -62,13 +62,14 @@
...
+                                  [('consul_svc_status', str),
+                                   ('consul_motr_proc_status', str)])
 
-MotrProcStatusLocalRemote = NamedTuple('MotrProcStatusLocalRemote', [(
-                                'motr_proc_status_local', ObjHealth),
-                                ('motr_proc_status_remote', ObjHealth)])
+MotrProcStatusLocalRemote = NamedTuple('MotrProcStatusLocalRemote',
```
Comment (mssawant) [inline hax/hax/util.py:69]: Same as above.

File: hax/hax/util.py [hunk 10]
```diff
@@ -2137,11 +2139,11 @@ def processEventToObjHealth(self, proc_type: m0HaProcessType,
 
     def service_health_to_m0dstatus_update(self, proc_fid: Fid,
                                            svc_health: ObjHealth):
-        ev = ConfHaProcess(chp_event=self.objHealthToProcessEvent(svc_health),
-                           chp_type=int(
-                                m0HaProcessType.M0_CONF_HA_PROCESS_M0D),
-                           chp_pid=0,
-                           fid=proc_fid)
+        ev = ConfHaProcess(
+            chp_event=self.objHealthToProcessEvent(svc_health),
```
Comment (mssawant) [inline hax/hax/util.py:2143]: Please avoid making unrelated changes. If there's a regression then it becomes difficult to map the changes to the problem.

Comment (mssawant) [reply hax/hax/util.py:2143]: Also less changes, less time to review.

File: cfgen/cfgen (hunk 1 above)
Comment (seagate-sarang-sawant) [reply cfgen/cfgen:46]: This was falke8 error indentation existing in this statement. Code remains the same.

File: hax/hax/util.py (hunk 8 above)
Comment (seagate-sarang-sawant) [reply hax/hax/util.py:65]: @mssawant  Yes following pep8 style. To keep the indentation like the previous statement, the list argument's [ bracket was brought down. If I pull "[" up then the rest of array elements will be moved like this.
faultToleranceInfo = NamedTuple('faultToleranceInfo', [
    ('max_drives_per_node', int),
    ('max_drives_per_ctrl', int),
    ('min_drives_per_node', int)])

File: cfgen/cfgen (hunk 2 above)
Comment (seagate-sarang-sawant) [reply cfgen/cfgen:273]: changed name from hndlr to handler.

**Review by d-nayak — APPROVED** [review]

File: cfgen/cfgen (hunk 2 above)
Comment (seagate-sarang-sawant) [reply cfgen/cfgen:273]: Fixed.

File: cfgen/cfgen (hunk 4 above)
Comment (seagate-sarang-sawant) [reply cfgen/cfgen:477]: Fixed.

File: cfgen/cfgen (hunk 6 above)
Comment (seagate-sarang-sawant) [reply cfgen/cfgen:569]: Fixed.

File: cfgen/cfgen (hunk 7 above)
Comment (seagate-sarang-sawant) [reply cfgen/cfgen:747]: Yes an additional info. Fixed.

File: hax/hax/util.py (hunk 9 above)
Comment (seagate-sarang-sawant) [reply hax/hax/util.py:69]: Fixed.

File: hax/hax/util.py (hunk 8 above)
Comment (seagate-sarang-sawant) [reply hax/hax/util.py:65]: Fixed.

File: cfgen/cfgen (hunk 3 above)
Comment (seagate-sarang-sawant) [reply cfgen/cfgen]: Fixed.

**Comment by mssawant** [comment]
@seagate-sarang-sawant, so now do you see the info messages in `kubectl logs` of hax container? Can you post the output please.

**Review by mssawant — APPROVED** [review]
Looks good in general. Please post the output of `kubectl logs` as requested. Thank you.

**Comment by seagate-sarang-sawant** [comment]
Kubectl logs for cortx-hax, cortx-setup containers  

[root@ssc-vm-rhev4-2619 ~]# kubectl logs -f -c cortx-hax cortx-data-g0-0
2022-08-18 12:42:25,231 [INFO] Starting Hare services
2022-08-18 12:42:25,298 [INFO] Entering logrotate_generic at line 161 in file /opt/seagate/cortx/hare/lib/python3.6/site-packages/hare_mp/main.py
2022-08-18 12:42:25,299 [INFO] Entering get_log_dir at line 697 in file /opt/seagate/cortx/hare/lib/python3.6/site-packages/hare_mp/main.py
2022-08-18 12:42:25,378 [INFO] Leaving get_log_dir
2022-08-18 12:42:25,379 [INFO] Leaving logrotate_generic
2022-08-18 12:42:25,385 [INFO] Entering start_hax_and_consul_without_systemd at line 415 in file /opt/seagate/cortx/hare/lib/python3.6/site-packages/hare_mp/main.py
2022-08-18 12:42:25,386 [INFO] Entering get_config_dir at line 705 in file /opt/seagate/cortx/hare/lib/python3.6/site-packages/hare_mp/main.py
2022-08-18 12:42:25,449 [INFO] Leaving get_config_dir
2022-08-18 12:42:25,449 [INFO] Entering get_log_dir at line 697 in file /opt/seagate/cortx/hare/lib/python3.6/site-packages/hare_mp/main.py
2022-08-18 12:42:25,509 [INFO] Leaving get_log_dir
2022-08-18 12:42:25,509 [INFO] Entering _start_consul at line 240 in file /opt/seagate/cortx/hare/lib/python3.6/site-packages/hare_mp/main.py
2022-08-18 12:42:25,572 [INFO] Entering Utils.get_local_hostname at line 91 in file /opt/seagate/cortx/hare/lib/python3.6/site-packages/hare_mp/utils.py
2022-08-18 12:42:25,572 [INFO] Entering Utils.get_hostname at line 79 in file /opt/seagate/cortx/hare/lib/python3.6/site-packages/hare_mp/utils.py
2022-08-18 12:42:25,572 [INFO] Leaving Utils.get_hostname
2022-08-18 12:42:25,572 [INFO] Leaving Utils.get_local_hostname
2022-08-18 12:42:25,573 [INFO] Entering ConsulStarter._execute at line 66 in file /opt/seagate/cortx/hare/lib/python3.6/site-packages/hare_mp/consul_starter.py
2022-08-18 12:42:30,613 [INFO] Leaving _start_consul
2022-08-18 12:42:30,613 [INFO] Entering _start_hax at line 299 in file /opt/seagate/cortx/hare/lib/python3.6/site-packages/hare_mp/main.py
2022-08-18 12:42:30,614 [INFO] Entering HaxStarter._execute at line 54 in file /opt/seagate/cortx/hare/lib/python3.6/site-packages/hare_mp/hax_starter.py
2022-08-18 12:42:30,614 [INFO] Leaving _start_hax


[root@ssc-vm-rhev4-2619 ~]# kubectl logs -c cortx-setup  cortx-data-g0-0
log: cortx_setup config apply -f yaml:///etc/cortx/solution/cluster.yaml -c 'yaml:///etc/cortx/cluster.conf'
2022-08-18 12:40:59 cortx_setup [19]: INFO [config_apply] Applying config yaml:///etc/cortx/solution/cluster.yaml
log: cortx_setup config apply -f yaml:///etc/cortx/solution/config.yaml -c 'yaml:///etc/cortx/cluster.conf'
2022-08-18 12:41:01 cortx_setup [42]: INFO [config_apply] Applying config yaml:///etc/cortx/solution/config.yaml
2022-08-18 12:41:01 cortx_setup [42]: WARNING [save] Release key cortx>common>release>name is missing or has incorrect value.
2022-08-18 12:41:01 cortx_setup [42]: INFO [save] Adding key "cortx>common>release>name" and value "CORTX" in confstore.
2022-08-18 12:41:01 cortx_setup [42]: WARNING [save] Release key cortx>common>release>version is missing or has incorrect value.
2022-08-18 12:41:01 cortx_setup [42]: INFO [save] Adding key "cortx>common>release>version" and value "2.0.0-7581" in confstore.
log: cortx_setup cluster bootstrap -c 'yaml:///etc/cortx/cluster.conf'
2022-08-18 12:41:15 cortx_setup [43]: INFO [_validate_provisioning_status] Recent phase for this node is None and recent status is None. Performing deployment on this node.
2022-08-18 12:41:15 cortx_setup [43]: INFO [cluster_deploy] Starting cluster bootstrap on 7ba8990a85d38508ca9bac4ff516d3ea:cortx-data-g0-0
2022-08-18 12:41:16 cortx_setup [43]: INFO [_provision_components] /opt/seagate/cortx/utils/bin/utils_setup post_install --config yaml:///etc/cortx/cluster.conf --services all
2022-08-18 12:41:17 cortx_setup [43]: INFO [_provision_components] /opt/seagate/cortx/motr/bin/motr_setup post_install --config yaml:///etc/cortx/cluster.conf --services io
2022-08-18 12:41:18 cortx_setup [43]: INFO [_provision_components] /opt/seagate/cortx/hare/bin/hare_setup post_install --config yaml:///etc/cortx/cluster.conf --services all
2022-08-18 12:41:20 cortx_setup [43]: INFO [_provision_components] /opt/seagate/cortx/utils/bin/utils_setup prepare --config yaml:///etc/cortx/cluster.conf --services all
2022-08-18 12:41:21 cortx_setup [43]: INFO [_provision_components] /opt/seagate/cortx/motr/bin/motr_setup prepare --config yaml:///etc/cortx/cluster.conf --services io
2022-08-18 12:41:21 cortx_setup [43]: INFO [_provision_components] /opt/seagate/cortx/hare/bin/hare_setup prepare --config yaml:///etc/cortx/cluster.conf --services all
2022-08-18 12:41:32 cortx_setup [43]: INFO [_provision_components] /opt/seagate/cortx/utils/bin/utils_setup config --config yaml:///etc/cortx/cluster.conf --services all
2022-08-18 12:41:32 cortx_setup [43]: INFO [_provision_components] /opt/seagate/cortx/motr/bin/motr_setup config --config yaml:///etc/cortx/cluster.conf --services io
2022-08-18 12:41:33 cortx_setup [43]: INFO [_provision_components] /opt/seagate/cortx/hare/bin/hare_setup config --config yaml:///etc/cortx/cluster.conf --services all
2022-08-18 12:41:48 cortx_setup [43]: INFO [_provision_components] /opt/seagate/cortx/utils/bin/utils_setup init --config yaml:///etc/cortx/cluster.conf --services all
2022-08-18 12:41:50 cortx_setup [43]: INFO [_provision_components] /opt/seagate/cortx/motr/bin/motr_setup init --config yaml:///etc/cortx/cluster.conf --services io
2022-08-18 12:41:51 cortx_setup [43]: INFO [_provision_components] /opt/seagate/cortx/hare/bin/hare_setup init --config yaml:///etc/cortx/cluster.conf --services all
2022-08-18 12:42:23 cortx_setup [43]: INFO [cluster_deploy] Finished cluster bootstrap on 7ba8990a85d38508ca9bac4ff516d3ea:cortx-data-g0-0


[root@ssc-vm-rhev4-2619 ~]# kubectl logs -c cortx-setup  cortx-ha-f9748b9f-g4zzc
log: cortx_setup config apply -f yaml:///etc/cortx/solution/cluster.yaml -c 'yaml:///etc/cortx/cluster.conf'
2022-08-18 12:40:36 cortx_setup [19]: INFO [config_apply] Applying config yaml:///etc/cortx/solution/cluster.yaml
log: cortx_setup config apply -f yaml:///etc/cortx/solution/config.yaml -c 'yaml:///etc/cortx/cluster.conf'
2022-08-18 12:40:38 cortx_setup [42]: INFO [config_apply] Applying config yaml:///etc/cortx/solution/config.yaml
2022-08-18 12:40:39 cortx_setup [42]: WARNING [save] Release key cortx>common>release>name is missing or has incorrect value.
2022-08-18 12:40:39 cortx_setup [42]: INFO [save] Adding key "cortx>common>release>name" and value "CORTX" in confstore.
2022-08-18 12:40:39 cortx_setup [42]: WARNING [save] Release key cortx>common>release>version is missing or has incorrect value.
2022-08-18 12:40:39 cortx_setup [42]: INFO [save] Adding key "cortx>common>release>version" and value "2.0.0-7581" in confstore.
log: cortx_setup cluster bootstrap -c 'yaml:///etc/cortx/cluster.conf'
2022-08-18 12:41:17 cortx_setup [43]: INFO [_validate_provisioning_status] Recent phase for this node is None and recent status is None. Performing deployment on this node.
2022-08-18 12:41:17 cortx_setup [43]: INFO [cluster_deploy] Starting cluster bootstrap on 7d9ee991ae6ce628a0e10294bdebaa6b:cortx-ha
2022-08-18 12:41:17 cortx_setup [43]: INFO [_provision_components] /opt/seagate/cortx/utils/bin/utils_setup post_install --config yaml:///etc/cortx/cluster.conf --services all
2022-08-18 12:41:19 cortx_setup [43]: INFO [_provision_components] /opt/seagate/cortx/ha/bin/ha_setup post_install --config yaml:///etc/cortx/cluster.conf --services all
2022-08-18 12:41:21 cortx_setup [43]: INFO [_provision_components] /opt/seagate/cortx/utils/bin/utils_setup prepare --config yaml:///etc/cortx/cluster.conf --services all
2022-08-18 12:41:21 cortx_setup [43]: INFO [_provision_components] /opt/seagate/cortx/ha/bin/ha_setup prepare --config yaml:///etc/cortx/cluster.conf --services all
2022-08-18 12:41:22 cortx_setup [43]: INFO [_provision_components] /opt/seagate/cortx/utils/bin/utils_setup config --config yaml:///etc/cortx/cluster.conf --services all
2022-08-18 12:41:23 cortx_setup [43]: INFO [_provision_components] /opt/seagate/cortx/ha/bin/ha_setup config --config yaml:///etc/cortx/cluster.conf --services all
2022-08-18 12:42:24 cortx_setup [43]: INFO [_provision_components] /opt/seagate/cortx/utils/bin/utils_setup init --config yaml:///etc/cortx/cluster.conf --services all
2022-08-18 12:42:25 cortx_setup [43]: INFO [_provision_components] /opt/seagate/cortx/ha/bin/ha_setup init --config yaml:///etc/cortx/cluster.conf --services all
2022-08-18 12:42:26 cortx_setup [43]: INFO [cluster_deploy] Finished cluster bootstrap on 7d9ee991ae6ce628a0e10294bdebaa6b:cortx-ha


[root@ssc-vm-rhev4-2619 ~]# kubectl logs -c cortx-setup  cortx-client-0
log: cortx_setup config apply -f yaml:///etc/cortx/solution/cluster.yaml -c 'yaml:///etc/cortx/cluster.conf'
2022-08-18 12:40:30 cortx_setup [19]: INFO [config_apply] Applying config yaml:///etc/cortx/solution/cluster.yaml
log: cortx_setup config apply -f yaml:///etc/cortx/solution/config.yaml -c 'yaml:///etc/cortx/cluster.conf'
2022-08-18 12:40:32 cortx_setup [42]: INFO [config_apply] Applying config yaml:///etc/cortx/solution/config.yaml
2022-08-18 12:40:32 cortx_setup [42]: WARNING [save] Release key cortx>common>release>name is missing or has incorrect value.
2022-08-18 12:40:32 cortx_setup [42]: INFO [save] Adding key "cortx>common>release>name" and value "CORTX" in confstore.
2022-08-18 12:40:32 cortx_setup [42]: WARNING [save] Release key cortx>common>release>version is missing or has incorrect value.
2022-08-18 12:40:32 cortx_setup [42]: INFO [save] Adding key "cortx>common>release>version" and value "2.0.0-7581" in confstore.
log: cortx_setup cluster bootstrap -c 'yaml:///etc/cortx/cluster.conf'
2022-08-18 12:41:17 cortx_setup [43]: INFO [_validate_provisioning_status] Recent phase for this node is None and recent status is None. Performing deployment on this node.
2022-08-18 12:41:17 cortx_setup [43]: INFO [cluster_deploy] Starting cluster bootstrap on c962d1686b3c5f268f5cfe967e45ab6b:cortx-client-0
2022-08-18 12:41:17 cortx_setup [43]: INFO [_provision_components] /opt/seagate/cortx/utils/bin/utils_setup post_install --config yaml:///etc/cortx/cluster.conf --services all
2022-08-18 12:41:19 cortx_setup [43]: INFO [_provision_components] /opt/seagate/cortx/motr/bin/motr_setup post_install --config yaml:///etc/cortx/cluster.conf --services motr_client
2022-08-18 12:41:20 cortx_setup [43]: INFO [_provision_components] /opt/seagate/cortx/hare/bin/hare_setup post_install --config yaml:///etc/cortx/cluster.conf --services all
2022-08-18 12:41:22 cortx_setup [43]: INFO [_provision_components] /opt/seagate/cortx/utils/bin/utils_setup prepare --config yaml:///etc/cortx/cluster.conf --services all
2022-08-18 12:41:23 cortx_setup [43]: INFO [_provision_components] /opt/seagate/cortx/motr/bin/motr_setup prepare --config yaml:///etc/cortx/cluster.conf --services motr_client
2022-08-18 12:41:24 cortx_setup [43]: INFO [_provision_components] /opt/seagate/cortx/hare/bin/hare_setup prepare --config yaml:///etc/cortx/cluster.conf --services all
2022-08-18 12:41:34 cortx_setup [43]: INFO [_provision_components] /opt/seagate/cortx/utils/bin/utils_setup config --config yaml:///etc/cortx/cluster.conf --services all
2022-08-18 12:41:35 cortx_setup [43]: INFO [_provision_components] /opt/seagate/cortx/motr/bin/motr_setup config --config yaml:///etc/cortx/cluster.conf --services motr_client
2022-08-18 12:41:35 cortx_setup [43]: INFO [_provision_components] /opt/seagate/cortx/hare/bin/hare_setup config --config yaml:///etc/cortx/cluster.conf --services all
2022-08-18 12:41:49 cortx_setup [43]: INFO [_provision_components] /opt/seagate/cortx/utils/bin/utils_setup init --config yaml:///etc/cortx/cluster.conf --services all
2022-08-18 12:41:50 cortx_setup [43]: INFO [_provision_components] /opt/seagate/cortx/motr/bin/motr_setup init --config yaml:///etc/cortx/cluster.conf --services motr_client
2022-08-18 12:41:51 cortx_setup [43]: INFO [_provision_components] /opt/seagate/cortx/hare/bin/hare_setup init --config yaml:///etc/cortx/cluster.conf --services all
2022-08-18 12:42:21 cortx_setup [43]: INFO [cluster_deploy] Finished cluster bootstrap on c962d1686b3c5f268f5cfe967e45ab6b:cortx-client-0
